pkg/services/authn/clients/api_key.go

```diff
@@ -170,50 +182,28 @@ func (s *APIKey) ResolveIdentity(ctx context.Context, orgID int64, typ claims.Id
 }
 
 func (s *APIKey) Hook(ctx context.Context, identity *authn.Identity, r *authn.Request) error {
-	id, exists := s.getAPIKeyID(ctx, identity, r)
-
-	if !exists {
+	if r.GetMeta(metaKeySkipLastUsed) != "" {
 		return nil
 	}
 
-	go func(apikeyID int64) {
+	go func(keyID string) {
 		defer func() {
 			if err := recover(); err != nil {
 				s.log.Error("Panic during user last seen sync", "err", err)
 			}
 		}()
-		if err := s.apiKeyService.UpdateAPIKeyLastUsedDate(context.Background(), apikeyID); err != nil {
-			s.log.Warn("Failed to update last use date for api key", "id", apikeyID)
-		}
-	}(id)
-
-	return nil
-}
 
-func (s *APIKey) getAPIKeyID(ctx context.Context, id *authn.Identity, r *authn.Request) (apiKeyID int64, exists bool) {
-	internalId, err := id.GetInternalID()
-	if err != nil {
-		s.log.Warn("Failed to parse ID from identifier", "err", err)
-		return -1, false
-	}
-
-	if id.IsIdentityType(claims.TypeAPIKey) {
-		return internalId, true
-	}
-
-	if id.IsIdentityType(claims.TypeServiceAccount) {
-		// When the identity is service account, the ID in from the namespace is the service account ID.
-		// We need to fetch the API key in this scenario, as we could use it to uniquely identify a service account token.
-		apiKey, err := s.getAPIKey(ctx, getTokenFromRequest(r))
+		id, err := strconv.ParseInt(keyID, 10, 64)
 		if err != nil {
-			s.log.Warn("Failed to fetch the API Key from request")
-			return -1, false
+			s.log.Warn("Failed to update last use date for api key", "id", keyID)
```

Comment:
This perhaps a real error and should return early? Log message should perhaps be different? Now it will continue and call `s.apiKeyService.UpdateAPIKeyLastUsedDate` with id=0. `err` is also lost/not logged

Reply:
Yes good catch. This _should_ never happen but if it does we want to have a proper message and include the error ofc